sdk/typespec/typespec_client_core/src/http/models/etag.rs

```diff
@@ -4,16 +4,22 @@
 use serde::{Deserialize, Serialize};
 use std::{fmt, str::FromStr};
 
+use crate::http::headers::HeaderValue;
+
 /// Represents an ETag for versioned resources.
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub struct Etag(String);
 
-impl<T> From<T> for Etag
-where
-    T: Into<String>,
-{
-    fn from(t: T) -> Self {
-        Self(t.into())
+// Implementation for common string types
+impl From<&str> for Etag {
+    fn from(s: &str) -> Self {
+        Self(s.to_string())
+    }
+}
+
+impl From<String> for Etag {
+    fn from(s: String) -> Self {
+        Self(s)
```

Comment:
Given `<String as Into>::into(self)` is basically a no-op (moved, but probably elided in release), why was the old impl so bad?